### ISSUE TITLE
connect to the cluster and using CCDT

### ISSUE THREAD

**Issue by pspratihar** [dsuch/pymqi#268]
Can you please add in your html page two more type of code :
(1) where python code connects by using the cluster name istead of the qmgr name
(2) python code connects using the CCDT file.

Thanks so much

**Comment by SeyfSV** [comment]
At this moment pymqi does support CCDT file :(

**Comment by JoshMcMQ** [comment]
I think it works as long as you leave off host, port,  channel.   It will look in all the standard places for the CCDT.

**Comment by Skyler-Altol** [comment]
Can confirm what JoshMcMQ is saying. I have personally done this for an application at work where I used a JSON CCDT to handle automatic reconnects and for easier readability (binary CCDTs can't be read with a text editor).

**Comment by SeyfSV** [comment]
Yes, it works with MQCONN, but as I can see it doesn't always work with MQCONNX. 

@pspratihar 
Your questions are not specifc to pymqi, and pymqi does not have any parameters related to CCDT.

**Comment by JoshMcMQ** [comment]
It works with MQCONNX just as well if the host,  port,  and channel are not provided, just don't pass MQCD.

**Comment by SeyfSV** [comment]
Yes, but pymqi does not have MQCONNX function. It is in pymqe.

**Comment by JoshMcMQ** [comment]
You just do this:

```
cd = None
qmgr.connect_with_options(MQDetails[EnvStore.QMGR],
                                  user=credentials[EnvStore.USER],
                                  password=credentials[EnvStore.PASSWORD],
                                  opts=options, cd=cd, sco=sco)
```

By setting cd to None it will work with CCDT.

See these examples on IBM's github:

https://github.com/ibm-messaging/mq-dev-patterns/tree/master/Python

**Comment by chughts** [comment]
The fix to enable CCDT was in #116 in June 2019

**Comment by SeyfSV** [comment]
Ok :)

**Comment by JoshMcMQ** [comment]
Technically that fix was to allow passing user and password via MQCSP while using a CCDT for the connection details.   Before this fix you could do it by calling the mqccred exit from the CCDT, see https://stackoverflow.com/questions/52883106/how-to-connect-to-ibm-mq-in-python-with-ccdt-file-userid-and-password
